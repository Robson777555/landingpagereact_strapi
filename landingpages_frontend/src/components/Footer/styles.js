import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../../components/TextComponent/styles';
import { Container as SectionContainer } from '../SectionContainer/styles';

export const Container = styled.footer`
  /* eslint-disable-next-line no-unused-vars */
  ${({ theme }) => css`
    font-family:${theme.fonts.family.default};
    text-align: center;
    font-size: ${theme.fonts.sizes.small};
    border-top: 0.1rem solid ${theme.colors.mediumGray};
    
    a{
      color: inherit;
      text-decoration: none;
      
    }
    & ${TextComponent} {
      font-size: ${theme.fonts.sizes.small};
    }

    & ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }
  `}
  
`;
