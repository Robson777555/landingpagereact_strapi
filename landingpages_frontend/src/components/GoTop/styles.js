import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    bottom: 2rem;
    right: 2rem;
    border-radius: 50%;
    z-index: 6;
    opacity: 0.7;
    transition: all 0.3s ease;
    cursor: pointer;

    /* Melhorias para mobile */
    @media (max-width: 768px) {
      width: 3.5rem;
      height: 3.5rem;
      bottom: 1.5rem;
      right: 1.5rem;
    }

    &:hover {
      opacity: 1;
      transform: scale(1.05);
    }

    /* Garante que seja clic√°vel em mobile */
    -webkit-tap-highlight-color: transparent;
    touch-action: manipulation;
  `}
`;
