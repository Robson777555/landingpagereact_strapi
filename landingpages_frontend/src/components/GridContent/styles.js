import styled, { css } from 'styled-components';

export const Container = styled.div`
  /* eslint-disable-next-line no-unused-vars */
  ${({ theme }) => css`
    font-family:${theme.fonts.family.default}; 
    text-align: center;
    max-width: 58rem;
    margin: 0 auto;
    

  `}
`;

export const html = styled.div`
  /* eslint-disable-next-line no-unused-vars */
  ${({ theme }) => css`
    font-family:${theme.fonts.family.default}; 
    margin:${theme.spacings.xhuge} 0;
    text-align: center;
    
    pre {
      display: inline-block;
      
      margin: 0 auto;
    }

  `}
`;
