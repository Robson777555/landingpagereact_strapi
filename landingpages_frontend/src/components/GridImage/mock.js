export default {
  background: false,
  title: 'My grid 2',
  description: 'Tecnologias usadas na Landing Page:',
  grid: [
    {
      altText: 'Teste 1',
      srcImg:
        'https://storage.googleapis.com/medium-feed.appspot.com/images%2F9353691196%2F2f49650ed42b1-Para-que-e-usado-o-JavaScript.jpg?r=1',
    },
    {
      altText: 'Teste 2',
      srcImg:
        'https://devsagaz.com.br/content/images/2023/05/Ekran-Resmi-2019-11-18-18.08.13.png?r=2',
    },
    {
      altText: 'Teste 3',
      srcImg:
        'https://mms.businesswire.com/media/20200520005179/en/792790/23/Logo.WhiteBackground.jpg?r=3',
    },
    {
      altText: 'Teste 4',
      srcImg:
        'https://upload.wikimedia.org/wikipedia/commons/d/d9/Node.js_logo.svg?r=4',
    },
    {
      altText: 'Teste 5',
      srcImg:
        'https://blog.codapp.com.br/wp-content/uploads/2022/05/capa.jpg?r=5',
    },
    {
      altText: 'Teste 6',
      srcImg:
        'https://learn.microsoft.com/pt-br/shows/visual-studio-code/media/visual-studio-code-banner-image.jpg?r=6',
    },
  ],
};
