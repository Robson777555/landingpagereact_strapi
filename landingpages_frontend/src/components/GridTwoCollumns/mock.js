export default {
  title: 'Grid Two Collumns',
  text: 'As três tecnolgias básicas para o front end são essas.',
  srcImg: 'public/assets/images/javascript.svg',
  background: false,
};
