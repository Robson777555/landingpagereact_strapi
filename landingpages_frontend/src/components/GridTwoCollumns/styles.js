import styled, { css } from 'styled-components';
import { Heading } from '../Heading/Heading';

export const Container = styled.div`
  ${({ theme, background, darkMode }) => css`
    display: grid;
    grid-template-columns: 1fr 1fr;
    align-items: center;
    gap: ${theme.spacings.large};
    width: 100%;
    max-width: 1200px;
    margin: 0 auto;
    padding: ${theme.spacings.large};
    background: ${
      background
        ? darkMode
          ? theme.colors.darkBg
          : theme.colors.white
        : 'transparent'
    };
    color: ${darkMode ? theme.colors.white : theme.colors.primaryColor};
    transition: all 300ms ease-in-out;

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
      text-align: center;
      gap: ${theme.spacings.medium};
    }

    @media ${theme.media.lteSmall} {
      padding: ${theme.spacings.medium} ${theme.spacings.small};
    }

    @media ${theme.media.iphone} {
      padding: ${theme.spacings.small} ${theme.spacings.xsmall};
      gap: ${theme.spacings.small};
    }

    /* Regra específica para 325px de largura e até 852px de altura */
    @media (max-width: 325px) and (max-height: 852px) {
      padding: ${theme.spacings.xsmall} ${theme.spacings.xxsmall || '0.4rem'};
      gap: ${theme.spacings.xsmall};
    }

    ${Heading} {
      color: ${darkMode ? theme.colors.white : theme.colors.black};
      margin-bottom: ${theme.spacings.medium};
      font-size: ${theme.fonts.sizes.xxlarge};
      font-weight: 700;
      text-transform: uppercase;
      line-height: 1.2;
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;

      @media ${theme.media.lteMedium} {
        font-size: ${theme.fonts.sizes.xlarge};
        margin-bottom: ${theme.spacings.small};
        white-space: normal;
      }

      @media ${theme.media.lteSmall} {
        font-size: ${theme.fonts.sizes.large};
        margin-bottom: ${theme.spacings.small};
      }

      @media ${theme.media.iphone} {
        font-size: ${theme.fonts.sizes.large};
        margin-bottom: ${theme.spacings.xsmall};
        white-space: nowrap;
        letter-spacing: -0.5px;
        padding: 0 ${theme.spacings.xsmall};
       
      }

      @media (max-width: 325px) and (max-height: 852px) {
        font-size: ${theme.fonts.sizes.medium};
        margin-bottom: ${theme.spacings.xsmall};
        letter-spacing: -0.3px;
        background: red;
      }
    }

    
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.default};
  `};
`;

export const TextComponent = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.default};
  `};
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;

    @media ${theme.media.iphone} {
      padding: 0 ${theme.spacings.xsmall};
    }

    @media (max-width: 325px) and (max-height: 852px) {
      padding: 0 ${theme.spacings.xxsmall || '0.4rem'};
    }
  `}
`;

export const Image = styled.img`
  ${({ theme, darkMode }) => css`
    width: 100%;
    max-width: 400px;
    height: auto;
    border-radius: ${theme.spacings.xsmall};
    filter: ${darkMode ? 'brightness(0.85)' : 'none'};
    transition: filter 300ms ease-in-out;

    @media ${theme.media.lteMedium} {
      max-width: 350px;
    }

    @media ${theme.media.lteSmall} {
      max-width: 280px;
    }

    @media ${theme.media.iphone} {
      max-width: 220px;
      margin: 0 auto;
      display: block;
    }

    @media (max-width: 325px) and (max-height: 852px) {
      max-width: 180px;
    }
  `}
`;

export const ToggleButton = styled.button`
  ${({ theme }) => css`
    position: fixed;
    bottom: ${theme.spacings.medium};
    right: ${theme.spacings.medium};
    padding: ${theme.spacings.small} ${theme.spacings.medium};
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    border: none;
    border-radius: 5px;
    cursor: pointer;
    z-index: 1000;
    transition: all 300ms ease-in-out;

    &:hover {
      background: ${theme.colors.secondaryColor};
      transform: scale(1.05);
    }

    @media ${theme.media.iphone} {
      padding: ${theme.spacings.xsmall} ${theme.spacings.small};
      bottom: ${theme.spacings.small};
      right: ${theme.spacings.small};
    }

    @media (max-width: 325px) and (max-height: 852px) {
      padding: ${theme.spacings.xxsmall || '0.4rem'} ${theme.spacings.xsmall};
      bottom: ${theme.spacings.xsmall};
      right: ${theme.spacings.xsmall};
      font-size: ${theme.fonts.sizes.xsmall};
      
    }
  `}
`;
