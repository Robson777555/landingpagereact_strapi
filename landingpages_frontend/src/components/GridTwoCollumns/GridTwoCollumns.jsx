import P from 'prop-types';
import * as Styled from './styles';
import { SectionBackground } from '../SectionBackground/SectionBackground';
import { Heading } from '../Heading/Heading';
import { TextComponent } from '../TextComponent/TextComponent';

export const GridTwoCollumns = ({
  title,
  text,
  srcImg,
  background = false,
}) => {
  return (
    <SectionBackground background={background}>
      <Styled.Container>
        <Styled.TextContainer>
          <Heading uppercase colorDark={!background} as="h2">
            {title}
          </Heading>
          <TextComponent colorDark={!background}>{text}</TextComponent>
        </Styled.TextContainer>
        <Styled.Image src={srcImg} alt={title} />
      </Styled.Container>
    </SectionBackground>
  );
};

GridTwoCollumns.propTypes = {
  title: P.string.isRequired,
  text: P.string.isRequired,
  srcImg: P.string.isRequired,
  background: P.bool,
};
