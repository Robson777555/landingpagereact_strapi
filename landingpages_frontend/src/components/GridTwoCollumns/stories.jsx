import { GridTwoCollumns } from './GridTwoCollumns';
import mock from './mock';
export default {
  title: 'GridTwoCollumns',
  component: GridTwoCollumns,
  args: mock,
  argTypes: {
    children: { type: 'string' },
  },
};

export const Template = (args) => {
  return (
    <div>
      <GridTwoCollumns {...args} />
    </div>
  );
};
