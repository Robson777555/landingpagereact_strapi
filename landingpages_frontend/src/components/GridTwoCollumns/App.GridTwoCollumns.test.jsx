import { screen } from '@testing-library/react';
import { GridTwoCollumns } from './GridTwoCollumns';
import { renderTheme } from '../../styles/render-theme';
import { describe, it, expect } from '@jest/globals';
import mock from './mock';
describe('<GridTwoCollumns />', () => {
  it('should render two columns grid', () => {
    const { container } = renderTheme(<GridTwoCollumns {...mock} />);
    expect(screen.getByRole('heading')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
