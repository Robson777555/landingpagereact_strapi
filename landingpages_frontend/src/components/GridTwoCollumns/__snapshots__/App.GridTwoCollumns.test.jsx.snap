// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridTwoCollumns /> should render two columns grid 1`] = `
.c4 {
  color: #0A1128;
  background: transparent;
  padding: 1rem;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c2 {
  display: grid;
  grid-template-columns: 1fr 1fr;
  align-items: center;
  gap: 3.2rem;
  width: 100%;
  max-width: 1200px;
  margin: 0 auto;
  padding: 3.2rem;
  background: transparent;
  color: #0A1128;
  transition: all 300ms ease-in-out;
}

.c2 [object Object] {
  color: #000000;
  margin-bottom: 2.4rem;
  font-size: 4.8rem;
  font-weight: 700;
  text-transform: uppercase;
  line-height: 1.2;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c3 {
  font-family: 'Open Sans',sans-serif;
}

.c6 {
  width: 100%;
  max-width: 400px;
  height: auto;
  border-radius: 0.8rem;
  filter: none;
  transition: filter 300ms ease-in-out;
}

.c0 {
  font-family: 'Open Sans',sans-serif;
  background: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: flex;
  align-items: center;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c5 {
  font-size: 2.4rem;
}

@media (max-width:768px) {
  .c4 {
    font-size: 4.0rem;
  }
}

@media (max-width: 768px) {
  .c2 {
    grid-template-columns: 1fr;
    text-align: center;
    gap: 2.4rem;
  }
}

@media (max-width: 480px) {
  .c2 {
    padding: 2.4rem 1.6rem;
  }
}

@media (max-width: 428px) {
  .c2 {
    padding: 1.6rem 0.8rem;
    gap: 1.6rem;
  }
}

@media (max-width:325px) and (max-height:852px) {
  .c2 {
    padding: 0.8rem 0.4rem;
    gap: 0.8rem;
  }
}

@media (max-width: 768px) {
  .c2 [object Object] {
    font-size: 4.0rem;
    margin-bottom: 1.6rem;
    white-space: normal;
  }
}

@media (max-width: 480px) {
  .c2 [object Object] {
    font-size: 3.2rem;
    margin-bottom: 1.6rem;
  }
}

@media (max-width: 428px) {
  .c2 [object Object] {
    font-size: 3.2rem;
    margin-bottom: 0.8rem;
    white-space: nowrap;
    letter-spacing: -0.5px;
    padding: 0 0.8rem;
  }
}

@media (max-width:325px) and (max-height:852px) {
  .c2 [object Object] {
    font-size: 2.4rem;
    margin-bottom: 0.8rem;
    letter-spacing: -0.3px;
    background: red;
  }
}

@media (max-width: 768px) {
  .c6 {
    max-width: 350px;
  }
}

@media (max-width: 480px) {
  .c6 {
    max-width: 280px;
  }
}

@media (max-width: 428px) {
  .c6 {
    max-width: 220px;
    margin: 0 auto;
    display: block;
  }
}

@media (max-width:325px) and (max-height:852px) {
  .c6 {
    max-width: 180px;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <h2
            class="c4"
          >
            Grid Two Collumns
          </h2>
          <div
            class="c5"
          >
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, quos.
          </div>
        </div>
        <img
          alt="Grid Two Collumns"
          class="c6"
          src="public/assets/images/javascript.svg"
        />
      </div>
    </div>
  </div>
</div>
`;
